### PR TITLE
NMS-15518: Simplify the installation docs

### PR DESCRIPTION
**Pull request by mfuhrmann**
### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://opennms.atlassian.net/) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the Jira link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the Jira issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* **If this code is likely to affect the UI, did you name your branch with `-smoke` in it to trigger smoke tests?**
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-15518

### CODE REVIEW

**Comment by mfuhrmann** [comment]
What I've done so far in the `core` section. 

1. Moved to reference section
* receive-snmp-traps
* install-jdbc-driver
* selinux
2. Also added a `related topics` section with those 3 references
3. Applied the encrypted database credentials way and dropped the clear text version. The idea here is to avoid letting people do the clear text way and after they are finished letting them know there is a better way :tada:
4. Renamed the title `Basic Horizon Deployment` to `OpenNMS {page-component-title}`

File: docs/modules/deployment/pages/core/getting-started.adoc [hunk 1]
```diff
@@ -30,9 +30,9 @@ endif::[]
 * Internet access to download the installation packages.
 * A working DNS server, and a localhost and server name that resolve properly.
 * A system user with administrative permissions (`sudo`) to perform installation.
-* A SELinux policy that permits binding to the ICMP service (if you are using an operating system that includes SELinux).
+* If in usage: A SELinux policy that permits binding to the ICMP service.
```
Comment (Bonrob2) [inline docs/modules/deployment/pages/core/getting-started.adoc:33]:
```suggestion
* If in use: A SELinux policy that permits binding to the ICMP service.
```

**Review by Bonrob2 — COMMENTED** [review]
Just one minor change. I think the restructuring works well. Thank you!

File: docs/modules/deployment/pages/core/debian-ubuntu/initialize-core.adoc [hunk 2]
```diff
@@ -1,25 +1,36 @@
...
                     database-name="template1"
                     class-name="org.postgresql.Driver"
                     url="jdbc:postgresql://localhost:5432/template1"
-                    user-name="postgres"<4>
-                    password="** YOUR-POSTGRES-PASSWORD **" /><5>
+                    user-name="${scv:postgres-admin:username}"<4>
+                    password="${scv:postgres-admin:password}" /><5>
 ----
 
 <1> Set the database name {page-component-title} should use.
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/debian-ubuntu/initialize-core.adoc:36]:
```suggestion
----
<1> Set the database name that {page-component-title} should use.
```

File: docs/modules/deployment/pages/core/debian-ubuntu/initialize-core.adoc [hunk 3]
```diff
@@ -47,7 +58,7 @@ During setup, the permissions for `net.ipv4.ping_group_range` are set permanentl
 Some system kernels do not honor this setting, and in those cases, additional configuration is required to allow binding to privileged ports.
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/debian-ubuntu/initialize-core.adoc:58]:
```suggestion
Some system kernels do not honor this setting; in those cases, additional configuration is required to allow binding to privileged ports.
```

File: docs/modules/deployment/pages/core/debian-ubuntu/initialize-core.adoc [hunk 4]
```diff
@@ -47,7 +58,7 @@ During setup, the permissions for `net.ipv4.ping_group_range` are set permanentl
...
 We recommend creating a systemd overlay to add the necessary settings to the service configuration.
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/debian-ubuntu/initialize-core.adoc:59]:
```suggestion
We recommend creating a systemd overlay to add the necessary settings to the service configuration:
```

File: docs/modules/deployment/pages/core/debian-ubuntu/initialize-core.adoc [hunk 5]
```diff
@@ -47,7 +58,7 @@ During setup, the permissions for `net.ipv4.ping_group_range` are set permanentl
...
 
-Run `systemctl edit --full opennms.service` and add the following line to the `[Service]` section:
+Run `systemctl edit opennms.service` and add the following line to the `[Service]` section:
 
 [source, properties]
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/debian-ubuntu/initialize-core.adoc:63]:
```suggestion
+
[source, properties]
```

File: docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc [hunk 6]
```diff
@@ -1,25 +1,36 @@
 .Configure PostgreSQL database access
+
+It is possible to store the PostgreSQL credentials in the secure credentials vault.
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc:3]:
```suggestion
You can store your PostgreSQL credentials in the secure credentials vault.
```

File: docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc [hunk 7]
```diff
@@ -1,25 +1,36 @@
...
+To achieve this, use the `scvcli` command line utility to add the credentials, and reference these credentials in `opennms-datasources.xml`.
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc:4]:
```suggestion
To do this, use the `scvcli` command line utility to add the credentials, and reference these credentials in `opennms-datasources.xml`:
```

File: docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc [hunk 8]
```diff
@@ -1,25 +1,36 @@
...
+
+.Create encrypted credentials for PostgreSQL database connections:
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc:6]:
```suggestion
. Create encrypted credentials for the PostgreSQL database connections:
+
```

File: docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc [hunk 9]
```diff
@@ -1,25 +1,36 @@
...
 [source, console]
+----
+sudo -u opennms ${OPENNMS_HOME}/bin/scvcli set postgres opennms password <1>
+sudo -u opennms ${OPENNMS_HOME}/bin/scvcli set postgres-admin postgres password
+----
+<1> Replace both `password` instances with the actual passwords for your `opennms` and `postgres` database accounts.
+
 ----
 sudo -u opennms vi /opt/opennms/etc/opennms-datasources.xml
 ----
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc:16]:
```suggestion
. Edit `opennms-datasources.xml` with `vi`:
+
[source, console]
sudo -u opennms vi /opt/opennms/etc/opennms-datasources.xml
```

File: docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc [hunk 10]
```diff
@@ -1,25 +1,36 @@
...
 
-.Set credentials to access the PostgreSQL database
+.Set encrypted credentials to access the PostgreSQL database
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc:18]:
```suggestion
. Set encrypted credentials to access the PostgreSQL database:
+
```

File: docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc [hunk 11]
```diff
@@ -1,25 +1,36 @@
...
 [source, xml]
 ----
 <jdbc-data-source name="opennms"
                     database-name="opennms"<1>
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc:22]:
```suggestion
                    database-name="opennms" <1>
```

File: docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc [hunk 12]
```diff
@@ -1,25 +1,36 @@
...
                     class-name="org.postgresql.Driver"
                     url="jdbc:postgresql://localhost:5432/opennms"
-                    user-name="** YOUR-OPENNMS-USERNAME **"<2>
-                    password="** YOUR-OPENNMS-PASSWORD **" /><3>
+                    user-name="${scv:postgres:username}"<2>
+                    password="${scv:postgres:password}" /><3>
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc:26]:
```suggestion
                    user-name="${scv:postgres:username}" <2>
                    password="${scv:postgres:password}" /> <3>
```

File: docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc [hunk 13]
```diff
@@ -1,25 +1,36 @@
...
 
 <jdbc-data-source name="opennms-admin"
                     database-name="template1"
                     class-name="org.postgresql.Driver"
                     url="jdbc:postgresql://localhost:5432/template1"
-                    user-name="postgres"<4>
-                    password="** YOUR-POSTGRES-PASSWORD **" /><5>
+                    user-name="${scv:postgres-admin:username}"<4>
+                    password="${scv:postgres-admin:password}" /><5>
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc:33]:
```suggestion
                    user-name="${scv:postgres-admin:username}" <4>
                    password="${scv:postgres-admin:password}" /> <5>
```

File: docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc [hunk 14]
```diff
@@ -1,25 +1,36 @@
...
 ----
 
 <1> Set the database name {page-component-title} should use.
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc:36]:
```suggestion
----
<1> Set the database name that {page-component-title} should use.
```

File: docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc [hunk 15]
```diff
@@ -45,7 +56,7 @@ sudo /opt/opennms/bin/install -dis
 {page-component-title} runs as a non-root user, which requires having a Linux kernel greater than 3.10.
 If you run on an older kernel, and are unable to upgrade your OS, you need to assign CAP_NET_RAW capabilities:
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc:57]:
```suggestion
If you run on an older kernel and you cannot upgrade your OS, you need to assign `CAP_NET_RAW` capabilities:
```

File: docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc [hunk 16]
```diff
@@ -45,7 +56,7 @@ sudo /opt/opennms/bin/install -dis
...
 
-Run `systemctl edit --full opennms.service` and add the following line to the `[Service]` section:
+Run `systemctl edit opennms.service` and add the following line to the `[Service]` section:
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc:59]:
```suggestion
. Run `systemctl edit opennms.service` and add the following code to the `[Service]` section:
```

File: docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc [hunk 17]
```diff
@@ -45,7 +56,7 @@ sudo /opt/opennms/bin/install -dis
...
 
 [source, properties]
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/centos-rhel/initialize-core.adoc:61]:
```suggestion
+
[source, properties]
```

**Review by emarsh1995 — CHANGES_REQUESTED** [review]
Some changes for flow and language. Thanks!

**Review by emarsh1995 — CHANGES_REQUESTED** [review]
Some changes for flow and language. Thanks!

ETA: GitHub submitted my review twice for some reason; please disregard this one.

File: docs/modules/reference/pages/configuration/selinux.adoc [hunk 18]
```diff
@@ -0,0 +1,76 @@
+[[selinux-icmp]]
+=== SELinux and ICMP permissions
```
Comment (mmahacek) [inline docs/modules/reference/pages/configuration/selinux.adoc:2]:
```suggestion
= SELinux and ICMP permissions
```

File: docs/modules/reference/pages/configuration/receive-snmp-traps.adoc [hunk 19]
```diff
@@ -0,0 +1,44 @@
+[[receive-snmp-traps]]
+== Receive SNMP traps/informs
```
Comment (mmahacek) [inline docs/modules/reference/pages/configuration/receive-snmp-traps.adoc:2]:
```suggestion
= Receive SNMP traps/informs
```

**Review by mmahacek — COMMENTED** [review]
I haven't read too far, and want to see it again after @emarsh1995's suggestions are processed, but I am liking the changes overall.
This is targeting `develop` which won't hit the docs until v32.  As the info does apply to previous versions, this should probably target `foundation-2023` (@Bonrob2 to confirm).

**Comment by mfuhrmann** [comment]
Closing this one. I pushed the changes into foundation-2023 base here https://github.com/OpenNMS/opennms/pull/5930

**Pull request closed without merging**